airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py

```diff
@@ -3,22 +3,51 @@
 #
 
 from abc import ABC
-from typing import Any, Iterable, Mapping, MutableMapping, Optional
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional
 
 import pendulum
+from airbyte_cdk.models import SyncMode
 from airbyte_cdk.sources.streams import Stream
+from google.ads.googleads.errors import GoogleAdsException
 from google.ads.googleads.v8.services.services.google_ads_service.pagers import SearchPager
 
 from .google_ads import GoogleAds
 
 
+def parse_dates(stream_slice):
+    start_date = pendulum.parse(stream_slice["start_date"])
+    end_date = pendulum.parse(stream_slice["end_date"])
+    return start_date, end_date
+
+
+def get_date_params(start_date: str, time_zone=None, range_days: int = None, end_date: pendulum.datetime = None):
+    """
+    Returns `start_date` and `end_date` for the given stream_slice.
+    If (end_date - start_date) is a big date range (>= 1 month), it can take more than 2 hours to process all the records from the given slice.
+    After 2 hours next page tokens will be expired, finally resulting in page token expired error
+    Currently this method returns `start_date` and `end_date` with 15 days difference.
+    """
+
+    end_date = end_date or pendulum.yesterday(tz=time_zone)
+    start_date = pendulum.parse(start_date)
+    if start_date > pendulum.now():
+        return start_date.to_date_string(), start_date.add(days=1).to_date_string()
+    end_date = min(end_date, start_date.add(days=range_days))
+
+    # Fix issue #4806, start date should always be lower than end date.
+    if start_date.add(days=1).date() >= end_date.date():
+        return start_date.add(days=1).to_date_string(), start_date.add(days=2).to_date_string()
+    return start_date.add(days=1).to_date_string(), end_date.to_date_string()
```

Comment:
Could you please explain why one day is added to start day?

Reply:
The same here. It's existing logic.
https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-google-ads/source_google_ads/streams.py#L110